CRM/Contact/Form/Contact.php

```diff
@@ -1496,6 +1496,10 @@ private function buildLocationBlock(string $name): void {
       CRM_Contact_Form_Edit_OpenID::buildQuickForm($this);
       return;
     }
```

Comment:
@eileenmcnaughton why can't we generalize this code to : 
```php
$className = "CRM_Contact_Form_Edit_$name";
if (class_exists($className)) {
   //special handling for email
   $className::buildQuickForm($this);
   retrurn;
}
```

Reply:
@monishdeb the problem is that when we construct class & function names as strings it is really hard to figure out where things are being called from - so we might save a few lines of code but we make it impossible to find the callers for functions & when we change / remove the functions things break - ie as happened in #30196 because this caller was not identified due to the concatenated string being un-greppable